### PR TITLE
Up/Down arrow move across cells (and more keyboard shortcuts)

### LINKED ISSUE #241: More keyboard shortcuts!!
1. Add shortcuts to add/remove cells
2. Up/down arrows should transition across cells if the current selection is the first/last line
3. Keyboard shortcuts should be more discoverable - maybe a modal to show all possible shortcuts?

### PR DESCRIPTION
**Pull request by indig**
Resolves #241

### CODE REVIEW

File: polynote-frontend/polynote/cell.js [hunk 1]
```diff
@@ -255,6 +255,25 @@ export class CodeCell extends Cell {
 
         this.editor.getModel().cellInstance = this;
 
+        this.editor.onKeyDown((evt) => {
+            const currentPos = this.editor.getPosition();
+            const modelRange = this.editor.getModel().getFullModelRange();
+            if (evt.code === "ArrowUp") {
```
Comment (jeremyrsmith) [inline polynote-frontend/polynote/cell.js]: Might want to check that the selection is empty (i.e. start position == end position). I.e. if I'm holding shift and press down a bunch of times to select lines, I probably don't want to move to the next cell if I press down once too many. (Maybe I would want that if you could select stuff across multiple cells... but you can't!)

File: polynote-frontend/polynote/cell.js [hunk 2]
```diff
@@ -831,6 +850,16 @@ export class TextCell extends Cell {
             } else if (evt.key === 'PageUp') {
                 this.dispatchEvent(new AdvanceCellEvent(this.id, true));
             }
+        } else if (evt.key === 'ArrowUp') {
```
Comment (jeremyrsmith) [inline polynote-frontend/polynote/cell.js]: Same on these ones – might want to make sure the selection is empty.

**Comment by indig** [comment]
@jeremyrsmith completely revamped this PR. Hopefully not too boring 😉

File: polynote-frontend/polynote/cell.js [hunk 3]
```diff
@@ -855,4 +945,53 @@ export class TextCell extends Cell {
...
         super.applyEdits(edits);
     }
+
+    getContentNodes() {
+        return Array.from(this.editorEl.childNodes)
+            // there are a bunch of text nodes with newlines we don't care about.
+            .filter(node => !(node.nodeType === Node.TEXT_NODE && node.textContent === '\n'))
+    }
+
+    // Note: lines in contenteditable are inherently weird, don't rely on this for anything aside from beginning and end
```
Comment (indig) [inline polynote-frontend/polynote/cell.js]: @jeremyrsmith, after our conversation I was able to modify my existing code to get it to work reasonably well while still providing some semblance of position. I added this note to show that the lines, while (ostensibly) internally consistent, don't necessarily reflect anything you might experience as a user.

**Pull request merged**